st2common/st2common/models/db/stormbase.py

```diff
@@ -105,6 +107,26 @@ def to_python(self, value):
         return mongoescape.unescape_chars(value)
 
 
+class TagField(me.EmbeddedDocument):
+    """
+    To be attached to a db model object for the purpose of providing supplemental
+    information.
+    """
+    name = me.StringField()
```

Comment:
I assume you do this so you can get indices. This increases the storage cost of tags. Just bringing it up to see if anyone has better suggestions.